### PR TITLE
UPSTREAM: 21005: Use a different verb for delete collection

### PR DESCRIPTION
**Pull request by liggitt**
Related to #7172
Merged upstream, holding for rebase

@deads2k @derekwaynecarr

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by deads2k** [comment]
@liggitt you going to plumb this through to our roles?

**Comment by liggitt** [comment]
hadn't decided yet... do we want project admins having this or do we want to limit it to things like the namespace controller for now?

**Comment by deads2k** [comment]
> hadn't decided yet... do we want project admins having this or do we want to limit it to things like the namespace controller for now?

Yeah, if you can list and you can delete, you should be able to deletecollection.  I just don't want to write equivalence logic.

**Comment by liggitt** [comment]
updated bootstrap policy

File: pkg/cmd/server/bootstrappolicy/infra_sa_policy.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func init() {
 				// BuildDeleteController.PodManager (ControllerClient)
 				// BuildControllerFactory.buildDeleteLW
 				{
-					Verbs:     sets.NewString("get", "list", "create", "delete"),
+					Verbs:     sets.NewString("get", "list", "create", "delete", "deletecollection"),
```
Comment (deads2k) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: These don't actually use these powers yet, right?  Seems like we'd keep the list tight and make them update it when they use it.

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/infra_sa_policy.go]: sure, removed from sa roles

**Comment by liggitt** [comment]
comment addressed, tag but hold for rebase

**Comment by deads2k** [comment]
lgtm

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
copr error, [merge]

**Pull request merged**